src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -4094,5 +4103,10 @@ public void AssertNotDisposed()
             if (_disposeOne.Disposed)
                 ThrowObjectDisposed();
         }
+
+        public static void LimitNumberOfConcurrentlyRunningIndexes(int maxNumberOfConcurrentlyRunningIndexes)
```

Comment:
Can this be called concurrently with indexing? If so, we may end up with releasing when we didn't acuqire.

Reply:
No. Anyway moved the lock to `ServerStore` so this is no longer relevant